vms/evm/destination_client.go

```diff
@@ -29,6 +31,11 @@ const (
 	MaxPriorityFeePerGas = 2500000000 // 2.5 gwei
 )
 
+// Client interface wraps the ethclient.Client interface for mocking purposes.
+type Client interface {
+	ethclient.Client
```

Comment:
Can we mock the ethclient.Client directly?

Reply:
if we mock the interface directly from the imported pkg we have to use mock reflect, which I didn't see a way to still work with `go generate`, so prefer to leave as is.